src/ripple/app/tx/impl/ApplyContext.cpp

```diff
@@ -103,17 +103,16 @@ ApplyContext::checkInvariantsHelper(
                     std::shared_ptr <SLE const> const& before,
                     std::shared_ptr <SLE const> const& after)
                 {
-                    // Sean Parent for_each_argument trick
-                    (void)std::array<int, sizeof...(Is)>{
-                        {((std::get<Is>(checkers).
-                            visitEntry(index, isDelete, before, after)), 0)...}
-                    };
+                    (..., std::get<Is>(checkers).visitEntry(index, isDelete, before, after));
                 });
 
-            // Sean Parent for_each_argument trick (a fold expression with `&&`
-            // would be really nice here when we move to C++-17)
-            std::array<bool, sizeof...(Is)> finalizers {{
-                std::get<Is>(checkers).finalize(tx, result, fee, journal)...}};
+            // Note: do not replace this logic with a `...&&` fold expression.
+            // The fold expression will only run until the first check fails (it
+            // short-circuits). While the logic is still correct, the log
+            // message won't be. Every failed invariant should write to the log,
+            // not just the first one.
+            std::array<bool, sizeof...(Is)> finalizers{
```

Comment:
Can we use CTAD here to just say `std::array` and let the value type and size be deduced by the RHS?

Reply:
Doesn't work for me:
```
error: no viable constructor or deduction guide for deduction of template arguments of 'array'

Reply:
Hmm, perhaps the compilers on CI don't have full deduction guides implemented for all containers? As far as I can tell, it should work fine. Not a biggie, though. :)